splunk_eventgen/lib/plugins/output/httpevent.py

```diff
@@ -182,50 +181,82 @@ def _sendHTTPEvents(self, payload):
             headers = {}
             headers['Authorization'] = server['header']
             headers['content-type'] = 'application/json'
             try:
-                payloadsize = len(payloadstring)
-                # response = requests.post(url, data=payloadstring, headers=headers, verify=False)
-                self.active_sessions.append(
-                    self.session.post(url=url, data=payloadstring, headers=headers, verify=False))
+                session_info = list()
+                session_info.append(url)
+                session_info.append(self.session.post(url=url, data=payloadstring, headers=headers, verify=False))
+                self.active_session_info.append(session_info)
             except Exception as e:
-                self.logger.error("Failed for exception: %s" % e)
                 self.logger.error("Failed sending events to url: %s  sourcetype: %s  size: %s" %
-                                  (url, self.lastsourcetype, payloadsize))
-                self.logger.debug(
-                    "Failed sending events to url: %s  headers: %s payload: %s" % (url, headers, payloadstring))
-                raise e
+                                  (url, self.lastsourcetype, len(payloadstring)))
+    
+    def reset_count(self, url):
+        try:
+            self.config.httpeventServersCountdownMap['url'] = self.httpeventAllowFailureCount
+        except:
+            pass
+        
+    def remove_requets_target(self, url):
+        if isinstance(self.config.httpeventServers, str):
+            httpeventServers = json.loads(self.config.httpeventServers)
+        else:
+            httpeventServers = self.config.httpeventServers
+
+        # If url fail more than specified count, we completely remove it from the pool.
+        try:
+            countdown_map = self.config.httpeventServersCountdownMap
+        except:
+            self.config.httpeventServersCountdownMap = {}
+            for i, server_info in enumerate(self.serverPool):
+                # URL is in format: https://2.2.2.2:8088/services/collector
+                self.config.httpeventServersCountdownMap[server_info.get('url', '')] = self.httpeventAllowFailureCount
+            countdown_map = self.config.httpeventServersCountdownMap
+
+        for i, server_info in enumerate(httpeventServers.get('servers', [])):
+            target_url = '{}://{}:{}'.format(server_info.get('protocol', ''), server_info.get('address', ''), server_info.get('port', '')) 
+            if target_url in url:
+                if countdown_map[url] <= 0:
+                    del httpeventServers.get('servers')[i]
+                    self.logger.warning("Cannot reach {}. Removing from the server pool".format(url))
+                else:
+                    countdown_map[url] -= 1
+                    self.logger.debug("Cannot reach {}. Lowering countdown to {}".format(url, countdown_map[url]))
+
+        if isinstance(self.config.httpeventServers, str):
```

Comment:
`self.config.httpeventServers` => `httpeventServers ` ?

Reply:
so here you want keep type consistent for `self.config.httpeventServers`.  If it was `str`, keep `str` value and if it is `object`, keep `object` value.  why not apply this to `self._sample.httpeventServers`, previous value is `str` while `httpeventServers` is object and assigned to it.